components/nifosg/nifloader.cpp

```diff
@@ -1319,10 +1327,43 @@ namespace NifOsg
                 return nullptr;
             }
 
-            unsigned char* data = new unsigned char[pixelData->data.size()];
-            memcpy(data, pixelData->data.data(), pixelData->data.size());
+            const std::vector<unsigned char>& pixels = pixelData->data;
+            if (pixelData->fmt != Nif::NiPixelData::NIPXFMT_PAL8)
+            {
+                unsigned char* data = new unsigned char[pixels.size()];
```

Comment:
IIRC, in C++11 vectors have the `data()` method to convert them to array, so maybe `copy()` is not needed here.

Reply:
Oh, never mind, OSG does not copy data in `setImage()` itself, it just stores a reference to the input array.